lib/nats.js

```diff
@@ -1003,6 +992,57 @@ Client.prototype.processInbound = function() {
     }
 };
 
+/**
+ * Process server updates in info object
+ * @api internal
+ */
+Client.prototype.processServerUpdate = function() {
+  var client = this;
+  if (client.info.connect_urls && client.info.connect_urls.length > 0) {
```

Comment:
As it happens, the new server will now send its own URL(s) as part of that INFO protocol, otherwise this `if` would have prevented from updating the server pool if all servers disconnected from the cluster. (note that older servers are not sending updates when servers go away, but would also not send their own urls).